walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,64 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.sql.Time;
+import java.util.*;
 
 public final class FindMeetingQuery {
+  /**
+   * Returns the times when {@code request} could happen on a day containing {@code events}
+   * Time complexity of the algorithm is O(nlogn + mn), in which
+   * n is the number of {@code events}, and m is the number of attendees in {@code request}
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<TimeRange> availableTimes = new ArrayList<>();
+    // Handle invalid request duration and no attendees edge case in request
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return availableTimes;
```

Comment:
One way I like to make the code better to understand is to return `Collections.emptyList()` instead of using a variable that is initialized somewhere else. This makes the code easier to comprehend.